backend/controllers/user.controller.ts

```diff
@@ -396,11 +408,22 @@ export const removeUserByID = async (
   res: Response,
   next: NextFunction,
 ) => {
-  const { userID } = req.body;
+  const validUserId = deleteSchema.safeParse(req.body);
```

Comment:
To make this delete controller consistent, I suggest getting the userId that we want to remove from the req.params . We can use the IdParamSchema  like we use in the update controller below so we don't need a separate deleteSchema.

router.delete("/:id", removeUserByID);